File: who_likes_it/who_likes_it.py
"""https://www.codewars.com/kata/5266876b8f4bf2da9b000362"""
def likes(names):
    res = ""
    match names:
        case []:
            res = "no one likes this"
        case [x]:
            res = f"{x} likes this"
        case [x, y]:
            res = f"{x} and {y} like this"
        case [x, y, z]:
            res = f"{x}, {y} and {z} like this"
        case [x, y, *rest]:
            res = f"{x}, {y} and {len(rest)} others like this"
    return res

File: who_likes_it/who_likes_it.rs
fn likes(names: &[&str]) -> String {
    match names[..] {
        [] => String::from("no one likes this"),
        [x] => format!("{} likes this", x),
        [x, y] => format!("{} and {} like this", x, y),
        [x, y, z] => format!("{}, {} and {} like this", x, y, z),
        [x, y, rest @ ..] => format!("{}, {} and {} others like this", x, y, rest.len()),
    }
}
